cpp/src/aztec/honk/proof_system/verifier.cpp

```diff
@@ -0,0 +1,121 @@
+#include <cmath>
+#include <common/throw_or_abort.hpp>
+#include <plonk/proof_system/constants.hpp>
+#include "./verifier.hpp"
+#include "../../plonk/proof_system/public_inputs/public_inputs.hpp"
+#include <ecc/curves/bn254/fq12.hpp>
+#include <ecc/curves/bn254/pairing.hpp>
+#include <ecc/curves/bn254/scalar_multiplication/scalar_multiplication.hpp>
+#include <polynomials/polynomial_arithmetic.hpp>
+#include <math.h>
+
+using namespace barretenberg;
+using namespace honk::sumcheck;
+
+namespace honk {
+template <typename program_settings>
+Verifier<program_settings>::Verifier(std::shared_ptr<waffle::verification_key> verifier_key,
+                                     const transcript::Manifest& input_manifest)
+    : manifest(input_manifest)
+    , key(verifier_key)
+{}
+
+template <typename program_settings>
+Verifier<program_settings>::Verifier(Verifier&& other)
+    : manifest(other.manifest)
+    , key(other.key)
+    , commitment_scheme(std::move(other.commitment_scheme))
+{}
+
+template <typename program_settings> Verifier<program_settings>& Verifier<program_settings>::operator=(Verifier&& other)
+{
+    key = other.key;
+    manifest = other.manifest;
+    commitment_scheme = (std::move(other.commitment_scheme));
+    kate_g1_elements.clear();
+    kate_fr_elements.clear();
+    return *this;
+}
+
+/**
+* @brief This function verifies a Honk proof for given program settings.
+*
+* TODO(luke): Complete this description
+* @detail A Standard Honk proof contains the following:
+    Multilinear evaluations:
+        w_i(X),        i = 1,2,3
+        sigma_i(X),    i = 1,2,3
+        q_i(X),        i = 1,2,3,4,5
+        z_perm(X),
+        L_0(X),
+        id(X)
+
+    Univariate evaluations:
+        a_0 = Fold_{-r}^(0)(-r),
+        a_l = Fold^(l)(-r^{2^l}), i = 1,...,d-1
```

Comment:
Index is `l`. Also, this is just uniformly written as `a_l = Fold_{-r}^(l)(-r^{2^l}), i = 0,...,d-1`

Reply:
I don't think so - a_0 is an evaluation of the partially evaluated batch polynomial, whereas l = 1, ..., d-1 are not

Reply:
My bad -- I was assuming a typo in the writeup, but I was wrong about that.